pkg/services/oauth/oauth.go

```diff
@@ -56,34 +95,196 @@ func (s *service) Configure() (err error) {
 		return nil, fmt.Errorf("identity token validation failed: %w", err)
 	}
-	if res.ValidationResult == services.Invalid {
+	if context.contractValidationResult.ValidationResult == services.Invalid {
 		return nil, fmt.Errorf("identity validation failed")
 	}
+	// checks if the name from the login contract matches with the registered name of the issuer.
+	if err = s.validateActor(&context); err != nil {
+		return nil, err
+	}
+
+	// validate the endpoint in aud, according to RFC003 §5.2.1.6
+	// the aud field must have the identifier of the endpoint registered by the vendor of this node!
+	// this is needed to prevent relay attacks.
+	// todo: implement when services and endpoints in registry have been implemented (https://github.com/nuts-foundation/nuts-registry/issues/156)
+
+	// validate the legal base, according to RFC003 §5.2.1.7 if sid is present
+	if err = s.validateLegalBase(context.jwtBearerToken); err != nil {
+		return nil, err
+	}
 
-	accessToken, err := s.buildAccessToken(jwtBearerToken, res)
+	accessToken, err := s.buildAccessToken(&context)
 	if err != nil {
 		return nil, err
 	}
 
 	return &services.AccessTokenResult{AccessToken: accessToken}, nil
 }
 
+// checks if the name from the login contract matches with the registered name of the issuer.
+func (s *service) validateActor(context *validationContext) error {
+	if context.contractValidationResult.ContractAttributes[contract.LegalEntityAttr] != context.actorName {
+		return errors.New("legal entity mismatch")
+	}
+	return nil
+}
+
+// check the actor against the registry, according to RFC003 §5.2.1.3
+// we do this by getting the validation chain for the certificate in the x5c header and check the vendorID SAN from the root
+// with the vendorId of the actor. It returns the actor name which must match the login contract.
+func (s *service) validateIssuer(context *validationContext) error {
+	validationTime := time.Unix(context.jwtBearerToken.IssuedAt, 0)
+	var vendor core.PartyID
+
+	actorPartyID, err := core.ParsePartyID(context.jwtBearerToken.Issuer)
+	if err != nil {
+		return fmt.Errorf(errInvalidIssuerFmt, err)
+	}
+	actor, err := s.registry.OrganizationById(actorPartyID)
+	if err != nil {
+		return fmt.Errorf(errInvalidIssuerFmt, err)
+	}
+	chains, err := s.crypto.TrustStore().VerifiedChain(context.jwtBearerToken.SigningCertificate, validationTime)
+	if err != nil || len(chains) == 0 {
+		return fmt.Errorf(errInvalidIssuerFmt, err)
+	}
+
+	match := false
+	for _, chain := range chains {
+		root := chain[len(chain)-1]
+		vendor, err = cert.VendorIDFromCertificate(root)
+		if err != nil {
+			fmt.Errorf("no vendorID in SAN: %w", err)
+		}
+		if vendor.String() == actor.Vendor.String() {
+			match = true
+			break
+		}
+	}
+	if !match {
+		return errors.New("certificate from x5c is no sibling of actor signing certificate")
```

Comment:
In think the 'sibling' concept is not "in de geest van" the X.509 spec. Isn't it just "certificate is not issued by vendor CA certificate"? Also it does't necesarrily need to be a sibling if the vendor CA certificate is renewed?

Reply:
changed to: `the signing certificate from the actor registry entry and the certificate from the x5c header do not share a common vendor CA`